website/docs/guides/0620-integration-key-vault.md

```diff
@@ -0,0 +1,169 @@
+
+---
+
+## Authentication Preliminaries
+
+### Referencing Certificates on Linux
+
+Virtual Client is a .NET application. Certificates used on a Linux system must be X.509 certificates containing a private key (e.g., PKCS#12, *.pfx). Certificates must be installed in the expected location for the user running Virtual Client:
+
+- **Root:** `/root/.dotnet/corefx/cryptography/x509stores/my/`
+- **Specific User:** `/home/{user}/.dotnet/corefx/cryptography/x509stores/my/`
+
+Ensure the user has read/write access to this directory and the certificate files. Set permissions as follows:
+
+sudo chmod -R 700 /home/{user}/.dotnet/corefx/cryptography/x509stores/my/
+
+### Referencing Certificates on Windows
+
+Virtual Client will look for certificates in these stores:
+
+- **CurrentUser/Personal:** The personal certificate store for the current user.
+- **LocalMachine/Personal:** The local machine certificate store.
+
+It is recommended to install certificates into the user-specific store.
+
+---
+
+## Key Vault Integration
+
+Virtual Client supports referencing Azure Key Vault using both URI-style and connection string-style formats. The Key Vault can be used to retrieve secrets (e.g., API keys, passwords) and certificates for use in your workloads.
+
+### Supported Reference Styles
+
+#### URI-Style References
+
+You can specify the Key Vault endpoint as a URI, optionally with authentication parameters:
+
+- **Microsoft Entra ID/App with Certificate (by thumbprint):**
+
+``` bash
+--keyvault="https://myvault.vault.azure.net?cid={clientId}&tid={tenantId}&crtt={certificateThumbprint}"
+```
+
+- **Microsoft Entra ID/App with Certificate (by issuer and subject):**
+
+``` bash
+--keyvault="https://myvault.vault.azure.net?cid={clientId}&tid={tenantId}&crti={issuer}&crts={subject}"
+```
+
+- **Managed Identity:**
+
+``` bash
+--keyvault="https://myvault.vault.azure.net?miid={managedIdentityClientId}"
+```
+
+#### Connection String-Style References
+
+You can also use a connection string format:
+
+- **Microsoft Entra ID/App with Certificate (by thumbprint):**
+
+``` bash
+--keyvault="EndpointUrl=https://myvault.vault.azure.net;ClientId={clientId};TenantId={tenantId};CertificateThumbprint={certificateThumbprint}"
+```
+
+- **Microsoft Entra ID/App with Certificate (by issuer and subject):**
+
+``` bash
+--keyvault="EndpointUrl=https://myvault.vault.azure.net;ClientId={clientId};TenantId={tenantId};CertificateIssuer={issuer};CertificateSubject={subject}"
+```
+
+- **Managed Identity:**
+
+``` bash
+--keyvault="EndpointUrl=https://myvault.vault.azure.net;ManagedIdentityId={managedIdentityClientId}"
+```
+
+---
+
+## Command-Line Options
+
+The following options are available for Key Vault integration:
+
+- `--kv`, `--keyvault`, `--key-vault`, `--keyVault`, `--key-Vault`
```

Comment:
nit: you could just remove the upper case ones, it's ok we choose all lower case flags